### ISSUE TITLE
Linkens & Mirrorshield procs together

### ISSUE THREAD

**Issue by GGSHAH** [ValveSoftware/Dota2-Gameplay#23291]
### Item name

Linkens & Mirroshield

### Description

Hello Dota Team,
I Faced this bug twice when playing **Dark Willow** in a Turbo. Both of my items were triggered even only 1 target spell was casted. Attaching both clips in this issue.

Checked this in demo hero as well. Faced it most of the times randomly. I don't know the reason behind it yet. Would be good to resolve this asap otherwise it will ruin the late games for most of the players.

https://github.com/user-attachments/assets/50340452-db67-4fb7-9c84-5f2af8db29c8

https://github.com/user-attachments/assets/c5017f1d-1d1a-4c0e-8ed3-a617cbd1998c

### Example Match ID and Timestamp

8087515153 - 40:19 & 52:03

**Comment by GGSHAH** [comment]
>I think it would be weird if only 1 of them goes into CD? if so which one? mirror shield first?
In hero demo I did notice this behavior as well. Sometimes only Mirror shield would  pop first and then Linkens afterwards on second spell.  Behavior should be consistent imo.

I think mirror shield  should goes in CD first  followed by Linkens or vice versa. Otherwise, it defeats the purpose of buying Linkens in late  game ( wasting ~5k gold)

**Comment by commentator24** [comment]
are you sure you waited until both CD finished? both have different CD

and in regards to LC duel thats just normal. Mirror shield reflects like lotus.
the same reason why lotus gives double duel dmg to LC it's because LC cast duel and you cast duel, thus 2x duel dmg.

basically with mirror you block LC duel cast, but you still cast duel on him. better take mirror off it doesnt work vs duel

2nd video i think techie just cast hex, blocked by both linken and mirror. mirror reflect hex, but this is blocked by linken in his stash (techie facet). then he cast bloodthorn silence

**Comment by GGSHAH** [comment]
>are you sure you waited until both CD finished? both have different CD

Yes I did. It was a very random behavior. Although I'm not 100% sure but I think just my hunch that if you put Linkens before Mirrorshield in inventory, both of them goes into CD together.  However if you put Mirrorshield shield first, only Mirrorshield  goes into CD. Gave this theory few attempts, and it noticed this behavior

>2nd video i think techie just cast hex, blocked by both linken and mirror. mirror reflect hex, but this is blocked by linken in his stash (techie facet). then he cast bloodthorn silence

I went to try it out in demo for this purpose to see if behavior was consistent. But it wasn't.Although I still think Mirrorshield and Linkens to have separate triggers if they're put together. Otherwise I don't see any point keeping both of them in late game which kinda makes game very unbalanced.
For example, you're into playing a 50+ minute game and need Linkens very badly but it will be a wastage of  money if you buy it just to sell it few minutes later once you get tier5 Mirror shield. I don't think it should be like that

**Comment by commentator24** [comment]
yeah you're right if linken is picked up first, both linken+mirror trigger so only block 1 spell.

 if mirror shield picked up first, only mirror trigger then linken, so block 2 spells, thats nice to know
 
 They could make a temporary simple fix by typing in tooltip must drop and pick up linken to fix trigger both same time
 and hope they fix bouncing spells, or at least add ss shard effect for reflected shaman shackle
